plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/modules/SourceModule.java

```diff
@@ -653,11 +653,12 @@ public TokensList getClassToks(ICompletionState initialState, ICodeCompletionAST
                     TokensList completions = getCompletionsForBase(initialState, manager, c, j);
                     modToks.addAll(completions);
                 }
+                modToks.setLookingFor(LookingFor.LOOKING_FOR_INSTANCED_VARIABLE);
             }
         } catch (CompletionRecursionException e) {
             // let's return what we have so far...
+            modToks.setLookingFor(initialState.getLookingFor());
```

Comment:
Humm, `CompletionRecursionException` shouldn't really happen, so, I think this effectively removes that statement (in which case you can just remove it altogether).

I have a question here: if you just don't set anything (i.e.: remove the `modToks.setLookingFor(LookingFor.LOOKING_FOR_INSTANCED_VARIABLE);` as well as this line), does it also fix the case?

Reply:
Apparently removing the statement also worked fine. I already remove it in last commit as well as updating old tests that include "self" as expected completion